src/server/frontend_xwayland/xwayland_wm.cpp

```diff
@@ -536,18 +535,17 @@ void mf::XWaylandWM::handle_client_message(xcb_client_message_event_t *event)
             get_window_debug_string(event->window).c_str());
     }
 
-    auto const surface_iter = surfaces.find(event->window);
-    if (surface_iter == surfaces.end())
-        return;
-
-    auto const surface = surface_iter->second;
+    auto const surface = get_wm_surface(event->window);
 
-    if (event->type == xcb_atom.net_wm_moveresize)
-        handle_move_resize(surface, event);
-    else if (event->type == xcb_atom.net_wm_state)
-        handle_state(surface, event);
-    else if (event->type == xcb_atom.wl_surface_id)
-        handle_surface_id(surface, event);
+    if (surface)
```

Comment:
Can't we reduce the scope of surface to the `if`? Vis:
```c++
    if (auto const surface = get_wm_surface(event->window))
```

Reply:
Good idea. Done